examples/check_unitary_std.py

```diff
@@ -14,22 +14,41 @@
 and show that is the same as the 
 unmasked time-domain.
 """
-#{{{generate data with just noise with a phase cycling dimension and repeats dimension
 N = 1024
 n_repeats = 50
 signal_window = (-100,200) # wherever my "peak" shows up
+# {{{ we know how to write a masked mean or std only along 1 dimension, so 
+#     use numpy apply_along_axis to make it a function that works along 1
+#     dimension of multidimensional data
+def masked_mean_multi(x, axis=None):
+    "Calculates the mean of nan-masked data on a 1D axis"
+    assert axis is not None
+    def masked_mean(x):
+        "this only works for 1D data"
+        return np.mean(x[np.isfinite(x)])
+    return np.apply_along_axis(masked_mean,axis,x)
+def masked_var_multi(x,axis=None, var_has_imag = True):
+    "calculates the variance of nan-masked data along a 1D axis"
+    assert axis is not None
+    def masked_var(x):
+        "this only works for 1D data"
+        if var_has_imag: # take average of variance along real and image
+            return np.var(x[np.isfinite(x)], ddof=1)/2
+        else:
+            return np.var(x[np.isfinite(x)], ddof=1)
+    return np.apply_along_axis(masked_var,axis,x)
+# }}}
+#{{ {generate data with just noise with a phase cycling dimension and repeats dimension
```

Comment:
let's not spin up new files until we need to.  If we're not using these functions in more than one modules, then we don't need to.